client/transport.go

```diff
@@ -21,7 +21,9 @@ func (tf transportFunc) RoundTrip(req *http.Request) (*http.Response, error) {
 func resolveTLSConfig(transport http.RoundTripper) (*tls.Config, error) {
 	switch tr := transport.(type) {
 	case *http.Transport:
-		return tr.TLSClientConfig, nil
+		if tr.DialTLS != nil {
+			return tr.TLSClientConfig, nil
+		}
 	case transportFunc:
 		return nil, nil // detect this type for testing.
 	default:
```

Comment:
to fix the current janky issue, please do remove this default statement here and move `return nil, errTLSConfigUnavailable` out of the switch statement